libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -36,6 +41,76 @@ ExpressionStatement makePopExpressionStatement(langutil::SourceLocation const& _
 	}};
 }
 
+void removeEmptyCasesFromSwitch(Switch& _switchStmt)
+{
+	auto& cases = _switchStmt.cases;
+
+	bool hasNonEmptyDefault = boost::algorithm::any_of(
+		cases,
+		[](Case const& _case) { return !_case.value && !_case.body.statements.empty(); }
+	);
+
+	if (!hasNonEmptyDefault)
```

Comment:
There are lots of negations in this condition.

To simplify this whole process, we could do the following instead:

 - first remove empty default if present (in it's own function)
 - then remove empty cases if there is no default case.